OpenRA.Mods.Common/UtilityCommands/UpgradeRules.cs

```diff
@@ -934,7 +934,144 @@ internal static void UpgradeActors(int engineVersion, ref List<MiniYamlNode> nod
 		{
 			foreach (var node in nodes)
 			{
-				// Add rules here
+				// Fix RA building footprints to not use _ when it's not necessary
+				if (engineVersion < 20160530 && Game.ModData.Manifest.Mod.Id == "ra")
+				{
+					// Tesla Coil
+					if (node.Value.Value == "tsla")
```

Comment:
I don't quite see the point of these upgrade rules. With hard-coding things like this, it'll only work on the official ra mod, which - when you have the utility that contains these rules - already has been adapted. I think you can safely remove these, unless I'm missing something not obvious.